internal/provider/provider.go

```diff
@@ -93,8 +101,99 @@ func (pm *NuoDbaasProviderModel) GetSkipVerify() bool {
 	return os.Getenv(NUODB_CP_SKIP_VERIFY) == "true"
 }
 
-func (pm *NuoDbaasProviderModel) CreateClient() (*openapi.Client, error) {
-	return nuodbaas_client.NewApiClient(pm.GetUrlBase(), pm.GetUser(), pm.GetPassword(), pm.GetToken(), pm.GetSkipVerify())
+func (pm *NuoDbaasProviderModel) getHttpClient() *http.Client {
+	client := &http.Client{}
+	if pm.GetSkipVerify() {
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true, //nolint:gosec // Reduced security at the demand of the user.
+			},
+		}
+	}
+	return client
+}
+
+func (pm *NuoDbaasProviderModel) getAuthHeader() string {
+	if pm.GetToken() != "" {
+		return "Bearer " + pm.GetToken()
+	} else if pm.GetUser() != "" {
+		auth := pm.GetUser() + ":" + pm.GetPassword()
+		return "Basic " + base64.StdEncoding.EncodeToString([]byte(auth))
+	}
+	return ""
+}
+
+func (pm *NuoDbaasProviderModel) prepareRequest(req *http.Request) {
+	if req.Header.Get("Authorization") == "" {
+		if authHeader := pm.getAuthHeader(); authHeader != "" {
+			req.Header.Set("Authorization", authHeader)
+		}
+	}
+}
+
+func (pm *NuoDbaasProviderModel) CreateClient() (openapi.ClientInterface, error) {
+	return openapi.NewClient(pm.GetUrlBase(),
+		openapi.WithHTTPClient(pm.getHttpClient()),
+		openapi.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+			pm.prepareRequest(req)
+			return nil
+		}))
+}
+
+func (pm *NuoDbaasProviderModel) buildSseRequest(ctx context.Context, path string) (*http.Request, error) {
+	url := strings.TrimSuffix(pm.GetUrlBase(), "/") + "/" + strings.TrimPrefix(path, "/")
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+	pm.prepareRequest(req)
+	return req, nil
+}
+
+func (pm *NuoDbaasProviderModel) createSseClient(ctx context.Context) *sse.Client {
+	// Copy default SSE client and replace HTTP client
+	sseClient := *sse.DefaultClient
```

Comment:
Nit: This is a shallow copy. Why not creating a new `sse.Client` instance?

Reply:
Good point.